orchagent/routeorch.cpp

```diff
@@ -88,6 +89,36 @@ RouteOrch::RouteOrch(DBConnector *db, vector<table_name_with_pri_t> &tableNames,
 
     SWSS_LOG_NOTICE("Maximum number of ECMP groups supported is %d", m_maxNextHopGroupCount);
 
+    /* fetch the MAX_ECMP_MEMBER_COUNT and for voq platform, set it to 128 */
+    attr.id = SAI_SWITCH_ATTR_MAX_ECMP_MEMBER_COUNT;
+    status = sai_switch_api->get_switch_attribute(gSwitchId, 1, &attr);
+    SWSS_LOG_NOTICE("Switch Type: %s, Max ECMP Group Size supported: %d", gMySwitchType.c_str(), attr.value.u32);
+    
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_WARN("Failed to get switch attribute max ECMP Group size. rv:%d", status);
+    }
+    else
+    {
+        uint32_t maxEcmpGroupSize = attr.value.u32;
+        /*If the switch type is voq, and max Ecmp group size supported is greater or equal to 128, set it to 128 */
+        if (gMySwitchType == "voq" && maxEcmpGroupSize >= 128)
```

Comment:
What is the purpose of forcing to 128. This should be ideally based on SAI capability.

Reply:
Hi @prsunny, in the control plane the max ECMP is set to 128(64 via ebgp and 64 via ibgp learnt routes), the purpose of forcing to 128 here to make sure the control plane and dataplane are in sync

```
 address-family ipv4 unicast
  network 10.1.0.1/32
  network 192.0.0.2/32 route-map HIDE_INTERNAL
  redistribute connected route-map V4_CONNECTED_ROUTES
  neighbor AH_V4 activate
  neighbor AH_V4 soft-reconfiguration inbound
  neighbor AH_V4 maximum-prefix 64000 90 warning-only
  neighbor AH_V4 route-map FROM_AH_V4 in
  neighbor AH_V4 route-map TO_AH_V4 out
  neighbor RH_V4 activate
  neighbor RH_V4 soft-reconfiguration inbound
  neighbor RH_V4 maximum-prefix 64000 90 warning-only
  neighbor RH_V4 route-map FROM_RH_V4 in
  neighbor RH_V4 route-map TO_RH_V4 out
  neighbor VOQ_CHASSIS_V4_PEER activate
  neighbor VOQ_CHASSIS_V4_PEER addpath-tx-all-paths
  neighbor VOQ_CHASSIS_V4_PEER soft-reconfiguration inbound
  neighbor VOQ_CHASSIS_V4_PEER route-map FROM_VOQ_CHASSIS_V4_PEER in
  neighbor VOQ_CHASSIS_V4_PEER route-map TO_VOQ_CHASSIS_V4_PEER out
  maximum-paths 64
  maximum-paths ibgp 64
 exit-address-family
 !
```